crates/transaction-pool/src/maintain.rs

```diff
@@ -257,8 +265,14 @@ pub async fn maintain_transaction_pool<N, Client, P, St, Tasks>(
                     .queued_transactions()
                     .into_iter()
                     .filter(|tx| {
-                        // filter stale external txs
-                        tx.origin.is_external() && tx.timestamp.elapsed() > config.max_tx_lifetime
+                        // filter stale transactions based on config
+                        if config.no_exemptions {
+                            // if no exemptions, treat all transactions the same
+                            tx.timestamp.elapsed() > config.max_tx_lifetime
+                        } else {
+                            // only evict external transactions
+                            tx.origin.is_external() && tx.timestamp.elapsed() > config.max_tx_lifetime
+                        }
```

Comment:
we can simplify this a bit by combining the first `tx.origin.is_external()` with `config.no_exemptions`

`(config.no_exemptions || tx.origin.is_external())`